src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/IteratorAndAsyncCaptureWalker.cs

```diff
@@ -194,7 +194,12 @@ private void CaptureVariable(Symbol variable, SyntaxNode syntax)
             var type = (variable.Kind == SymbolKind.Local) ? ((LocalSymbol)variable).Type : ((ParameterSymbol)variable).Type;
             if (type.IsRestrictedType())
             {
-                (_lazyDisallowedCaptures ??= new MultiDictionary<Symbol, SyntaxNode>()).Add(variable, syntax);
+                // Interpolated string builder variables are limited in lifetime, and will not be used in a context that
+                // actually needs to be saved to the closure.
+                if (variable is not InterpolatedStringBuilderLocalSymbol)
```

Comment:
Why can't we just let the machinery to figure this out.

Reply:
I'm honestly not sure what you mean by this. What machinery are you referring to?

---
In reply to: [610693222](https://github.com/dotnet/roslyn/pull/52422#discussion_r610693222) [](ancestors = 610693222)

Reply:
> I'm honestly not sure what you mean by this. What machinery are you referring to?

Given that "and will not be used in a context that actually needs to be saved to the closure", why do we need this special case at all? 

---
In reply to: [612033020](https://github.com/dotnet/roslyn/pull/52422#discussion_r612033020) [](ancestors = 612033020,610693222)

Reply:
I think this was due to the disposal mechanisms. As that's no longer present, the machinery no longer needs to be adjusted.

---
In reply to: [612764219](https://github.com/dotnet/roslyn/pull/52422#discussion_r612764219) [](ancestors = 612764219,612033020,610693222)